### ISSUE TITLE
requests reducer for useRequest

### ISSUE THREAD

**Issue by staydecent** [inputlogic/elements#77]

**Comment by staydecent** [comment]
```
import W from 'wasmuth'

// Makes clearing requests easier.

type Predicate = (string) => boolean

const CLEAR_REQUEST = 'CLEAR_REQUEST'
const CLEAR_REQUESTS = 'CLEAR_REQUESTS'

const clearRequest = (endpointOrUid: string): FluxStandardAction => {
  return { type: CLEAR_REQUEST, payload: { endpointOrUid } }
}

const clearRequests = (predicate: Predicate): FluxStandardAction => {
  return { type: CLEAR_REQUESTS, payload: { predicate } }
}

export const actions = {
  clearRequest,
  clearRequests
}

export function requestsReducer (action: FluxStandardAction, state: any) {
  const { type, payload } = action
  if (type === CLEAR_REQUEST) {
    return {
      ...state,
      requests: W.without([payload.endpointOrUid], state.requests)
    }
  } else if (type === CLEAR_REQUESTS) {
    const { predicate } = payload
    const matchedRequestKeys = W.filter(predicate, Object.keys(state.requests))
    return {
      ...state,
      requests: W.without(matchedRequestKeys, state.requests)
    }
  }
  return state
}
```

**Comment by staydecent** [comment]
Closed by #79
